pkg/metrics-server/metrics.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metric_server
+
+import (
+	"net/http"
+
+	apimetrics "k8s.io/apiserver/pkg/endpoints/metrics"
+	"k8s.io/apiserver/pkg/server/mux"
+	etcd3metrics "k8s.io/apiserver/pkg/storage/etcd3/metrics"
+	flowcontrolmetrics "k8s.io/apiserver/pkg/util/flowcontrol/metrics"
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+// DefaultMetrics installs the default prometheus metrics handler
+type DefaultMetrics struct {
+	registry metrics.KubeRegistry
+}
+
+// Install adds the DefaultMetrics handler
+func (m DefaultMetrics) Install(c *mux.PathRecorderMux) {
+	register()
+	c.HandleFunc("/metrics", func(w http.ResponseWriter, req *http.Request) {
+		legacyregistry.Handler().ServeHTTP(w, req)
+		metrics.HandlerFor(m.registry, metrics.HandlerOpts{}).ServeHTTP(w, req)
+	})
+}
+
+// register apiserver and etcd metrics
+func register() {
```

Comment:
@s-urbaniak @serathius as part of this effort, is there a plan to get rid of the metrics we were inheriting from the generic API server?

Reply:
My gut feeling is that this should be a separate effort, but maybe @serathius knows more :thinking:

Reply:
Metrics Server serves Metrics API, so apiserver metrics will be useful. I think we can remove etcd and flowcontrol metrics.